src/error.ts

```diff
@@ -60,12 +60,12 @@ export const GET_MORE_RESUMABLE_CODES = new Set<number>([
 ]);
 
 /** @public */
-export interface ErrorDescription {
+export interface ErrorDescription extends Document {
   message?: string;
   errmsg?: string;
   $err?: string;
   errorLabels?: string[];
-  [key: string]: any;
+  errInfo?: Document;
```

Comment:
I think removing `[key: string]: any` is lying about the type, because we still expect this description to contain arbitrary keys from the server (and we parse them in and assign them to the error in the MongoServerError class)

Reply:
missed the extends above, resolving